### PR TITLE
Angular: Unify Principal and Account Service

### PR DESCRIPTION
**Pull request by wmarques**
We have two services who manages the same things IMO: the user account. I think we should merge them into the `AccountService` which will provide the necessary functions.
I also removed the async `hasAnyAuthority`, I don't know it was async as we don't need any network calls.

-   [ ] [Travis tests](https://travis-ci.org/jhipster/generator-jhipster/pull_requests) are green
-   [ ] Tests are added where necessary
-   [ ] Documentation is added/updated where necessary
-   [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

### CODE REVIEW

File: generators/client/templates/angular/src/main/webapp/app/core/auth/account.service.ts.ejs [hunk 1]
```diff
@@ -18,19 +18,114 @@
...
+
+    fetch(): Observable<HttpResponse<Account>> {
         return this.http.get<Account>(SERVER_API_URL + '<%- apiUaaPath %>api/account', {observe : 'response'});
     }
 
     save(account: any): Observable<HttpResponse<any>> {
         return this.http.post(SERVER_API_URL + '<%- apiUaaPath %>api/account', account, {observe: 'response'});
     }
+
+    authenticate(identity) {
```
Comment (deepu105) [inline generators/client/templates/angular/src/main/webapp/app/core/auth/account.service.ts.ejs]: could you add tests for this methods?

**Comment by deepu105** [comment]
@jdubois seems like I do not have access to retry jobs on azure pipeline, could you give me access

**Comment by deepu105** [comment]
ye but I can't even restart the entire PR build

Thanks & Regards,
Deepu

File: generators/client/templates/angular/src/main/webapp/app/core/auth/account.service.ts.ejs (hunk 1 above)
Comment (wmarques) [reply generators/client/templates/angular/src/main/webapp/app/core/auth/account.service.ts.ejs]: Sure, I'll write some during the weekend

**Comment by wmarques** [comment]
@deepu105 it's done, I added some unit tests, tell me if that's ok for you :)

**Comment by wmarques** [comment]
@DanielFran good catch

**Comment by wmarques** [comment]
Finally green 🎉 I forgot to take care of your comment @DanielFran I'll do that asap and it should be good

**Comment by wmarques** [comment]
@DanielFran it's done !

**Pull request merged**